NetProbe/Utility/TestStorage.mm

```diff
@@ -0,0 +1,120 @@
+
+@implementation TestStorage
+
++ (NSArray*)get_tests{
+    if ([[NSUserDefaults standardUserDefaults] objectForKey:@"tests"]){
+        return [[NSUserDefaults standardUserDefaults] objectForKey:@"tests"];
+    }
+    return nil;
+}
+
++ (void)add_test:(NetworkMeasurement*)test{
+    [self checkTests];
+    NSMutableArray *cache = [[[NSUserDefaults standardUserDefaults] arrayForKey:@"tests"] mutableCopy];
+    [cache addObject:[NSKeyedArchiver archivedDataWithRootObject:test]];
+    [[NSUserDefaults standardUserDefaults] setObject:cache forKey:@"tests"];
+    [[NSUserDefaults standardUserDefaults] synchronize];
+}
+
++ (NSArray*)remove_test:(NSNumber*)test_id{
+    [self checkTests];
+    NSMutableArray *cache = [[[NSUserDefaults standardUserDefaults] arrayForKey:@"tests"] mutableCopy];
+    for (int i = 0; i < [cache count]; i++) {
+        NetworkMeasurement* test = [NSKeyedUnarchiver unarchiveObjectWithData:[cache objectAtIndex:i]];
+        if (test.test_id == test_id){
+            [self removeFile:test.json_file];
+            [self removeFile:test.log_file];
+            [cache removeObjectAtIndex:i];
+        }
+    }
+    [[NSUserDefaults standardUserDefaults] setObject:cache forKey:@"tests"];
+    [[NSUserDefaults standardUserDefaults] synchronize];
+    return cache;
+}
+
++ (NSArray*)remove_test_atindex:(long)index{
+    [self checkTests];
+    NSMutableArray *cache = [[[NSUserDefaults standardUserDefaults] arrayForKey:@"tests"] mutableCopy];
+    if ([cache count] > index) {
+        NetworkMeasurement* test = [NSKeyedUnarchiver unarchiveObjectWithData:[cache objectAtIndex:index]];
+        [self removeFile:test.json_file];
+        [self removeFile:test.log_file];
+        [cache removeObjectAtIndex:index];
+    }
+    [[NSUserDefaults standardUserDefaults] setObject:cache forKey:@"tests"];
+    [[NSUserDefaults standardUserDefaults] synchronize];
+    return cache;
+}
+
++ (void)checkTests{
+    if (![[NSUserDefaults standardUserDefaults] arrayForKey:@"tests"]){
+        NSMutableArray *cache = [[NSMutableArray alloc] init];
+        [[NSUserDefaults standardUserDefaults] setObject:cache forKey:@"tests"];
+        [[NSUserDefaults standardUserDefaults] synchronize];
+    }
+}
+
++ (NetworkMeasurement*)get_test_atindex:(long)index{
+    [self checkTests];
+    NSMutableArray *cache = [[[NSUserDefaults standardUserDefaults] arrayForKey:@"tests"] mutableCopy];
+    if ([cache count] > index) {
+        NetworkMeasurement* test = [NSKeyedUnarchiver unarchiveObjectWithData:[cache objectAtIndex:index]];
+        return test;
+    }
+    return nil;
+}
+
++ (void)set_completed:(NSNumber*)test_id {
+    [self checkTests];
+    NSMutableArray *cache = [[[NSUserDefaults standardUserDefaults] arrayForKey:@"tests"] mutableCopy];
+    for (int i = 0; i < [cache count]; i++) {
+        NetworkMeasurement* test = [NSKeyedUnarchiver unarchiveObjectWithData:[cache objectAtIndex:i]];
+        if ([test.test_id isEqualToNumber:test_id]){
```

Comment:
Here you use this construct to check for equality, above you use `test.test_id == test_id`... I am wondering (again this is ignorance!) if they are equivalent!